src/universal/redux/makeReducer.js

```diff
@@ -6,15 +6,41 @@ import authToken from './authDuck';
 import {reducer as storageReducer} from 'redux-storage-whitelist-fn';
 import storageMerger from 'universal/redux/storageMerger';
 
-const currentReducers = {
+const initialReducers = {
   authToken,
   cashay: cashayReducer,
   form: formReducer,
   notifications
 };
+const currentReducers = {...initialReducers};
 
 export default (newReducers) => {
+  /**
+   * Add the initial state of the new reducers to the initialReducers object. If
+   * we later revert the application to the initial state -- say to logout --
+   * we'll need these dynamically loaded initial states present so
+   * mapStateToProps can run one last time before the component unmounts:
+   */
+  Object.assign(initialReducers, newReducers);
```

Comment:
Actually, mutating initialReducers is _exactly_ what we want to do. We need the initial states of any dynamically loaded reducers added here to, or bad things happen in mSTP of dynamically loaded components just before they are unmounted.

If you'd like to see what I mean, try commenting this out and see what happens. I think it'll be fairly obvious...

Procedure:

1. Open console
1. Log in
1. Click gear in hub
1. Log out